packages/shared/lib/clients/orchestrator.ts

```diff
@@ -306,71 +331,101 @@ export class Orchestrator {
+                        }
+                    });
+                    res = webhookResult.mapError((e) => new NangoError('action_failure', e.payload ?? { error: e.message }));
+                    if (res.isErr()) {
+                        span.setTag('error', res.error);
+                    }
                 } catch (e: unknown) {
-                    const errorMsg = `Execute: Failed to parse input object ${JSON.stringify(input)} to JsonValue: ${stringifyError(e)}`;
-                    logger.error(errorMsg);
+                    const errorMsg = `Execute: Failed to parse input '${JSON.stringify(input)}': ${stringifyError(e)}`;
+                    const error = new NangoError('action_failure', { error: errorMsg });
+                    span.setTag('error', e);
+                    return Err(error);
+                } finally {
+                    span.finish();
                 }
-            }
+            } else {
+                const span = tracer.startSpan('execute.webhookTemporal', {
+                    tags: spanTags,
+                    ...(activeSpan ? { childOf: activeSpan } : {})
+                });
+                try {
+                    const temporal = await getTemporal();
+                    const webhookHandler = await temporal.workflow.execute('webhook', {
+                        taskQueue: WEBHOOK_TASK_QUEUE,
+                        workflowId,
+                        args: [
+                            {
+                                name: webhookName,
+                                parentSyncName: syncConfig.sync_name,
+                                nangoConnection: nangoConnectionWithoutCredentials,
+                                input,
+                                activityLogId
+                            }
+                        ]
+                    });
 
-            const temporal = await getTemporal();
-            const webhookHandler = await temporal.workflow.execute('webhook', {
-                taskQueue: WEBHOOK_TASK_QUEUE,
-                workflowId,
-                args: [
-                    {
-                        name: webhookName,
-                        parentSyncName: syncConfig.sync_name,
-                        nangoConnection: nangoConnectionWithoutCredentials,
-                        input,
-                        activityLogId
+                    const { error, response } = webhookHandler;
+                    if (error) {
+                        res = Err(error);
+                    } else {
+                        res = Ok(response);
                     }
-                ]
-            });
-
-            const { success, error, response } = webhookHandler;
+                    if (res.isErr()) {
+                        span.setTag('error', res.error);
+                    }
+                } catch (e: unknown) {
+                    span.setTag('error', e);
+                    throw e;
```

Comment:
it didn't used to throw, is it okay?

Reply:
Also curious about this throw

Reply:
it didn't have a local try-catch before. This is just re-throwing for it ot be handled by the main try-catch